src/Mod/Part/App/WireJoiner.cpp

```diff
@@ -0,0 +1,2440 @@
+                            showShape(*splitWire, "swire", iteration);
+                        }
+
+                        checkWireInfo(*newWire);
+                        break;
+                    }
+                    if (newWire) {
+                        ++iteration;
+                        break;
+                    }
+
+                    if (++idxV == (int)wireVertices.size())
+                        break;
+                    stack.emplace_back(vertexStack.size());
+                    ++stack.back().iEnd;
+                    vertexStack.push_back(wireVertices[idxV]);
+                    edgeSet.insert(wireVertices[idxV].edgeInfo());
+                    checkStack();
+                }
+
+                if (!newWire) {
+                    showShape(*beginInfo.wireInfo, "done", iteration);
+                    beginInfo.wireInfo->done = true;
+                    // If a wire is done, make sure all edges of this wire is
+                    // marked as done. This can also prevent duplicated wires.
+                    for (auto &v : beginInfo.wireInfo->vertices) {
+                        auto info = v.edgeInfo();
+                        if (!info->wireInfo) {
+                            info->wireInfo = beginInfo.wireInfo;
+                            continue;
+                        }
+                        else if (info->wireInfo->done)
+                            continue;
+                        auto otherWire = info->wireInfo;
+                        auto &otherWireVertices = info->wireInfo->vertices;
+                        if (info == otherWireVertices.front().edgeInfo()) {
+                            // About to change the first edge of the other wireInfo.
+                            // Try to find a new first edge for it.
+                            tmpVertices.clear();
+                            auto it = otherWireVertices.begin();
+                            tmpVertices.push_back(*it);
+                            for (++it;it!=otherWireVertices.end();++it) {
+                                if (it->edgeInfo()->wireInfo == otherWire)
+                                    break;
+                                tmpVertices.push_back(*it);
+                            }
+                            if (tmpVertices.size() != otherWireVertices.size()) {
+                                otherWireVertices.erase(otherWireVertices.begin(), it);
+                                otherWireVertices.insert(otherWireVertices.end(),
+                                        tmpVertices.begin(), tmpVertices.end());
+                            }
+                        }
+                        assertCheck(info != &beginInfo);
+                        info->wireInfo = beginInfo.wireInfo;
+                        checkWireInfo(*otherWire);
+                    }
+                    checkWireInfo(*beginInfo.wireInfo);
+                }
+            }
+        }
+    }
+
+    void exhaustTightBound()
+    {
```

Comment:
I hate to say this because I know how much work is involved, but this many nested loops creates a lot of cognitive complexity, so maybe refactor to multiple methods?   Or just resolve this; I'm okay either way.

Reply:
If I remember correctly also other Toponaming methods have been refactored to reduce cognitive complexity, therefor yes I could do the same here.

@chennes what do you think about that?

Reply:
Generally speaking yes: this should absolutely be refactored. The big question is whether you feel like the current tests are enough to make it comfortable to do so. If not, then it should be a project for a later date, preceded by adding more tests.

Reply:
Ok then I'll see if there are more tests needed and if not will refactor.

Thanks!